client/components/Project/AddProjectComponent.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+  <div>
+    <button class="add-btn" @click="showModal">
+      + Add Project
+    </button>
+    <Modal
+      v-show="show"
+      :hideModal="hideModal"
+    >
+      <h2>Add Project</h2>
+      <form @submit.prevent="submit">
+        <div class="field">
+          <label for="project-name">
+            Name
+          </label>
+          <input
+            id="name"
+            name="name"
+            type="text"
+            placeholder="6.1040 Final Project"
+            :value="fields.name"
+            @input="fields.name = $event.target.value"
+          />
+        </div>
+        <div class="field">
+          <label for="project-name">
+            Deadline
+          </label>
+          <input
+            id="name"
+            name="name"
+            type="date"
+            :min="minDate"
+            :value="fields.deadline"
+            @input="fields.deadline = $event.target.value"
+          />
+        </div>
+        <div class="field">
+          <label for="collaborators">
+            Collaborators
+          </label>
+          <div class="collaborators-list">
+            <div
+              v-for="(collaborator, i) in fields.collaborators"
+              class="collaborator"
+            >
+              <label>
+                ({{i + 1}})
+              </label>
+              <input
+                :id="i"
+                :name="i"
+                :value="fields.collaborators[i]"
+                placeholder="john.doe@gmail.com"
+                @input="fields.collaborators[i] = $event.target.value"
+              />
+              <button class="text-btn" @click="removeCollaborator(i)">
+                🗑
+              </button>
+            </div>
+            <button
+              class="text-btn"
+              @click="$event.preventDefault(); addCollaborator()"
+            >
+              + Add collaborator
+            </button>
+          </div>
+        </div>
+      </form>
+      <div class="actions">
+        <button class="invert" @click="hideModal">Cancel</button>
+        <button type="submit" @click="submit">Add</button>
+      </div>
+    </Modal>
+  </div>
+</template>
+
+<script>
+import moment from 'moment';
+import Modal from '@/components/common/Modal.vue';
+
+export default {
+  name: 'AddProjectComponent',
+  components: {Modal},
+  data() {
+    return {
+      show: false,
+      minDate: moment().format('YYYY-MM-DD'),
+      fields: {
+        name: '',
+        deadline: '',
+        collaborators: [''],
+      },
+    }
+  },
+  methods: {
+    hideModal() {
+      this.show = false;
+    },
+    showModal() {
+      this.show = true;
+    },
+    submit() {
+      console.log(this.fields);
+      this.hideModal();
+    },
+    addCollaborator() {
+      this.fields.collaborators.push('');
+    },
+    removeCollaborator(i) {
```

Comment:
Not sure if this is the reason but there is an issue when removing a collaborator that is not the last one. For example, in the picture, if you click delete for (1) or (2), it will close the modal window but preserve correct info, so the problem is just that it closes the modal window for some reason?
<img width="490" alt="image" src="https://user-images.githubusercontent.com/19766890/205453310-aa08c164-1cd6-4640-8de6-2adfe1e9354e.png">

Reply:
Whoops good catch! Fixed.